smart_card_connector_app/src/chrome-api-provider.js

```diff
@@ -308,11 +308,14 @@ function logUnexpectedConnectionState(state) {
  * chrome.smartCardProviderPrivate.ConnectionState. We only take the highest bit
  * of the value, since it contains all the needed information, e.g.
  * SCARD_NEGOTIABLE implies that SCARD_POWERED and SCARD_PRESENT are also set.
- * @param {number} state
+ * @param {number} pcscState
  * @returns {!chrome.smartCardProviderPrivate.ConnectionState}
  * @throws {Error} Throws error if unknown connection state is encountered.
  */
-function convertConnectionStateToEnum(state) {
+function convertConnectionStateToEnum(pcscState) {
+  // Upper 16 bits contains the number of events for the reader and don't have
+  // state flags.
+  state = pcscState & 0x0000FFFF;
```

Comment:
nit: This bitmask starts being repeated. Maybe introduce a constant for it (or for its negation)?

Reply:
thanks!